waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -130,56 +141,48 @@ seed projectDir seedName =
-dbExecuteTest projectDir =
-  let absSchemaPath = projectDir </> dbSchemaFileInProjectRootDir
-   in runPrismaCommandAsDbJob
-        projectDir
-        (const ["db", "execute", "--stdin", "--schema", SP.fromAbsFile absSchemaPath])
+dbExecuteTest projectRootDir =
+  let absSchemaPath = projectRootDir </> dbSchemaFileInProjectRootDir
+   in runPrismaCommandAsJob
+        projectRootDir
+        ["db", "execute", "--stdin", "--schema", SP.fromAbsFile absSchemaPath]
 
 -- | Runs `prisma studio` - Prisma's db inspector.
 runStudio :: Path' Abs (Dir ProjectRootDir) -> J.Job
-runStudio projectDir = runPrismaCommandAsDbJob projectDir $ \schema ->
-  ["studio", "--schema", SP.fromAbsFile schema]
-
-generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> (String, String) -> JobType -> J.Job
-generatePrismaClient projectDir prismaClientOutputDirEnv jobType =
-  runPrismaCommandAsJobWithExtraEnv jobType envVars projectDir $ \schema ->
-    ["generate", "--schema", SP.fromAbsFile schema]
+runStudio projectRootDir =
+  runPrismaCommandAsJob projectRootDir ["studio", "--schema", SP.fromAbsFile schema]
   where
-    envVars = [prismaClientOutputDirEnv]
+    schema = projectRootDir </> dbSchemaFileInProjectRootDir
 
-runPrismaCommandAsDbJob ::
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsDbJob projectDir makeCmdArgs =
-  runPrismaCommandAsJob J.Db projectDir makeCmdArgs
+generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> J.Job
+generatePrismaClient projectRootDir =
+  runPrismaCommandAsJob projectRootDir args
+  where
+    args =
+      [ "generate",
+        "--schema",
+        SP.fromAbsFile $ projectRootDir </> dbSchemaFileInProjectRootDir
+      ]
 
-runPrismaCommandAsJob ::
-  JobType ->
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsJob jobType projectDir makeCmdArgs =
-  runPrismaCommandAsJobWithExtraEnv jobType [] projectDir makeCmdArgs
+runPrismaCommandAsJob :: Path' Abs (Dir ProjectRootDir) -> [String] -> J.Job
+runPrismaCommandAsJob projectRootDir args =
+  runPrismaCommandAsJobWithExtraEnv [] projectRootDir args
 
 runPrismaCommandAsJobWithExtraEnv ::
-  JobType ->
   [(String, String)] ->
   Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
+  [String] ->
   J.Job
-runPrismaCommandAsJobWithExtraEnv jobType envVars projectDir makeCmdArgs =
-  runNodeCommandAsJobWithExtraEnv envVars serverDir (absPrismaExecutableFp projectDir) (makeCmdArgs schemaFile) jobType
+runPrismaCommandAsJobWithExtraEnv envVars projectRootDir args =
+  runNodeCommandAsJobWithExtraEnv envVars waspProjectDir (absPrismaExecutableFp waspProjectDir) args J.Db
   where
-    serverDir = projectDir </> serverRootDirInProjectRootDir
-    schemaFile = projectDir </> dbSchemaFileInProjectRootDir
+    waspProjectDir = projectRootDir </> waspProjectDirFromProjectRootDir
```

Comment:
This doesn't seem to be the cleanest way of specifying the Wasp project root as a working directory. Still, I choose to go with it because:
- It keeps the directory resolution in a single location.
- It ensures the simplest possible type signatures (i.e., no need of dragging two path arguments all the way down to the command)
- It requires the fewest changes.
- All functions above `runPrismaCommandAsJob` and `runPrismaCommandAsJobWithExtraEnv` really do need the path `ProjectRootDir`, so the "dirty part" is very localized.
- The signatures of `runPrismaCommandAsJobWithExtraEnv` and `runPrismaCommandAsJob` should specify the same `StrongPath` directory in their arguments, so it's either `WaspProjectDir` or `ProjectRootDir`. Going with the first one would require many changes and a lot of duplication.

Reply:
Looking at it, doesn't look like there is something very wrong with it. If you have these two pieces of information, `projectRootDir` and `waspProjectDirFromProjectRootDir`, why wouldn't you use them here?

I guess the real problem lies in the "dirtiness" of `waspProjectDirFromProjectRootDir`?

Ok, just to make sure I got this right: you were considering passing `ProjectRootDir` (what we have now) vs passing `WaspProjectDir` (what we could have in theory) instead?
And it feels better to pass the path that is higher above? Why would one be dirtier than the another?

To me, dirty thing seems to be `waspProjectDirFromProjectRootDir` at the moment, if that one is made clean, then this is also clean. The question is, can it be made clean in the form it is in now.

Btw I am trying to guess a bit here, because you seem to have an alternative approach in mind but don't describe it explicitly. If I got it wrong, please correct me.

Reply:
> Ok, just to make sure I got this right: you were considering passing ProjectRootDir (what we have now) vs passing WaspProjectDir (what we could have in theory) instead?

If we wanted to avoid getting one from the other using relative paths, we'd need to pass both `ProjectRootDir` and `WaspProjectDir` almost all the way down. If we don't care about using relative paths for getting one from the other, then I think this is the cleanest solution.

The part I found dirty was defining the `WaspProjectDir` in terms of `ProjectRootDir` (as opposed to sending it down). So, yes, defining `waspProjectDirFromProjectRootDir` felt like a hack.

I think the textbook approach is sending both directories all the way down, but it just seems unnecessarily cumbersome.

Why did you find `waspProjectDirFromProjectRootDir` dirty, if not for this reason?

Reply:
I don't think both pieces of information have to be passed down, I think it is ok to pass one piece and have another hardcoded if that makes sense for us (and it could/does). I don't see why that would be dirty.

What I didn't like is how `waspProjectDirFromProjectRootDir` is defined, and I commented on that in another comment. So we can close this one and continue there.

Reply:
We discussed live how to solve this, and agreed to move the `../../` next to definition of `out` and `build` dirs, as the fastest okish solution.